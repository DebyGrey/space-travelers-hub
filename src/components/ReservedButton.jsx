const ReservedButton = () => (
  <>
    <button type="button" className="bg-teal-500 text-sm font-bold text-white py-[0.1rem] px-[0.3rem] rounded mr-2">Reserved</button>
  </>
);
export default ReservedButton;
