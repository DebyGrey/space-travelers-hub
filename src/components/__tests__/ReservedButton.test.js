import { render, screen } from '@testing-library/react';
import ReservedButton from '../ReservedButton';

describe('ReservedButton', () => {
  test('renders the ReservedButton component', () => {
    render(<ReservedButton />);

    const buttonElement = screen.getByText('Reserved');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveClass('bg-teal-500');
    expect(buttonElement).toHaveClass('text-sm');
    expect(buttonElement).toHaveClass('font-bold');
    expect(buttonElement).toHaveClass('text-white');
    expect(buttonElement).toHaveClass('py-[0.1rem]');
    expect(buttonElement).toHaveClass('px-[0.3rem]');
    expect(buttonElement).toHaveClass('rounded');
    expect(buttonElement).toHaveClass('mr-2');
  });

  test("renders the ReservedButton component", () => {
    const { asFragment } = render(<ReservedButton />);

    expect(asFragment()).toMatchSnapshot();
  });
});
