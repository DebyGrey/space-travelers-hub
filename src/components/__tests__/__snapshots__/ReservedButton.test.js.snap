// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservedButton renders the ReservedButton component 1`] = `
<DocumentFragment>
  <button
    class="bg-teal-500 text-sm font-bold text-white py-[0.1rem] px-[0.3rem] rounded mr-2"
    type="button"
  >
    Reserved
  </button>
</DocumentFragment>
`;
